src/components/AppButton.mjs

```diff
@@ -0,0 +1,30 @@
+export default class AppButton extends HTMLElement {
+
+static get observedAttributes() {
+        return ['type']
+    }
+    
+    constructor() {
+        super()
+		const template = document.querySelector('#app-button').innerHTML
+        const shadow = this.attachShadow({mode: 'open'})
+        shadow.innerHTML = template
+        const elem = this.shadowRoot.querySelector('.change-type')
```

Comment:
Tip: You could also use `this.shadowRoot.querySelector('button')` since its the only button element in the template. No need to change it but just wanted you to know its possible

Reply:
Thank you for your feedback, will try it out and re-submit today. Thanks!

Reply:
That makes sense.